src/assets/TokenRatesController.ts

```diff
@@ -137,6 +192,22 @@ export class TokenRatesController extends BaseController<
     }, this.config.interval);
   }
 
+  set chainId(chainId: string) {
```

Comment:
We've been trying to move away from using setters for this kinda thing, and instead preferring methods. If this was a method (like `setChainId`) it's clear that setting it has a side-effect, and the caller can look to the method definition to see what it does. It might even have a nice JSDoc comment (which gets shown inline via intellisense on some editors).

But with a setter like this... it's ambiguous to the caller what effect setting this has. It might have a setter, or it might be referenced anywhere in this controller or in other modules, or both.

Reply:
Is there an example of the right way to do this?
I've been using it as a side-effect handler of `controller.config({...})`

Reply:
Ah, right, I forgot `configure` worked that way. Ah well. Maybe this is fine until switching to the new BaseControllerV2 API then. Not a huge deal.